From 8cf55453606a4cc2c41accbd7f34f85d83c344b1 Mon Sep 17 00:00:00 2001
From: Hongzhang Yang <Hongzhang.Yang@freescale.com>
Date: Tue, 13 Aug 2013 11:02:56 +0800
Subject: [PATCH 406/505] ENGR00274897 Improve VPU unit test

- Fix library link issue for some strict tool chain
  Move -lpthread to the end of link command line
- Prevent data type redefinition

Signed-off-by: Hongzhang Yang <Hongzhang.Yang@freescale.com>
---
 test/mxc_vpu_test/Makefile   | 2 +-
 test/mxc_vpu_test/vpu_test.h | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/test/mxc_vpu_test/Makefile b/test/mxc_vpu_test/Makefile
index a3a38c9..c8849d4 100644
--- a/test/mxc_vpu_test/Makefile
+++ b/test/mxc_vpu_test/Makefile
@@ -23,7 +23,7 @@ endif
 all: $(TARGET)
 
 $(TARGET): $(OBJ)
-	$(LINK) -lpthread -o $(TARGET) $(OBJ) ${LIBS} -lvpu -lipu -lrt
+	$(LINK) -o $(TARGET) $(OBJ) ${LIBS} -lvpu -lipu -lrt -lpthread
 	mkdir -p $(OBJDIR)
 	mv $(TARGET) $(OBJDIR)
 	cp autorun-vpu.sh $(OBJDIR)
diff --git a/test/mxc_vpu_test/vpu_test.h b/test/mxc_vpu_test/vpu_test.h
index 5c5b6ff..9730ce9 100644
--- a/test/mxc_vpu_test/vpu_test.h
+++ b/test/mxc_vpu_test/vpu_test.h
@@ -45,9 +45,9 @@ extern int vpu_test_dbg_level;
 	printf("[WARN]\t" fmt, ## arg);	\
 	} while (0)
 
-#ifdef u32
 #undef u32
-#endif
+#undef u16
+#undef u8
 typedef unsigned long u32;
 typedef unsigned short u16;
 typedef unsigned char u8;
-- 
1.8.0

